### PR TITLE
Follows feature

### PR DESCRIPTION
**Pull request by naid**
-this set of files contain the feature for following and unfollowing users

### CODE REVIEW

File: app/Follow.php [hunk 1]
```diff
@@ -19,4 +20,17 @@ public function followers()
     {
         return $this->hasMany(User::class, 'follower_id');
     }
+
+    public function addFollowee($id)
+    {
+        $this->follower_id = Auth::id();
```
Comment (nguyenthanhtung88) [inline app/Follow.php]: Don't use `auth` in Model, pass param from Controller is better

File: app/Follow.php [hunk 2]
```diff
@@ -19,4 +20,17 @@ public function followers()
     {
         return $this->hasMany(User::class, 'follower_id');
     }
+
+    public function addFollowee($id)
```
Comment (nguyenthanhtung88) [inline app/Follow.php]: This function should be `public function addFollowee($followerId, $followeeId)`

File: app/Follow.php [hunk 3]
```diff
@@ -19,4 +20,17 @@ public function followers()
...
+    {
+        $this->follower_id = Auth::id();
+        $this->followee_id = $id;
+        $this->save();
+    }
+
+    public function removeFollowee($id)
```
Comment (nguyenthanhtung88) [inline app/Follow.php]: This function shouldd be `public function removeFollowee($userid, $followeeId)`

File: app/Follow.php [hunk 4]
```diff
@@ -19,4 +20,17 @@ public function followers()
...
+    {
+        $fff = Follow::where('followee_id', $id)->where('follower_id', Auth::id());
```
Comment (nguyenthanhtung88) [inline app/Follow.php]: Variable `$fff` doesn't make sense

Comment (naid) [inline app/Follow.php]: ooopps, sorry. This was during debugging.

File: app/Http/Controllers/UserController.php [hunk 5]
```diff
@@ -96,4 +97,35 @@ public function updatePassword(Request $request)
...
+        $title = trans('common.users.list_users_title');
+
+        $users = User::all();
+        $follows = Follow::where('follower_id', $this->user->id)->lists('followee_id');
+
+        return view('users.list', [
+            'user' => $this->user,
+            'usersList' => $users,
+            'follows' => $follows->toArray(),
+            'title' => $title . $this->user->id,
```
Comment (nguyenthanhtung88) [inline app/Http/Controllers/UserController.php]: `title` param pass to view but not use

File: app/Http/Controllers/UserController.php [hunk 6]
```diff
@@ -96,4 +97,35 @@ public function updatePassword(Request $request)
...
 
+    public function listUsers()
+    {
+        $title = trans('common.users.list_users_title');
+
+        $users = User::all();
+        $follows = Follow::where('follower_id', $this->user->id)->lists('followee_id');
+
+        return view('users.list', [
+            'user' => $this->user,
```
Comment (nguyenthanhtung88) [inline app/Http/Controllers/UserController.php]: I saw you always have to pass variable `user` in views which extends `app` view.
Have you ever been using share data to all view.
https://laravel.com/docs/5.1/views#sharing-data-with-all-views
=> You just only write `view()->share('user', $user)` in your Base Controller, and all your views can use this variable

P/S: Maybe i forget suggest you in previous PR, i see in this controller you have to pass `$user` variable in index, edit, ...

File: resources/views/users/list.blade.php [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+                        <th class="col-md-3">&nbsp;</th>
+                        <th class="col-md-1">&nbsp;</th>
+                        <th class="col-md-2">Name</th>
+                        <th class="col-md-3">Email</th>
+                        <th class="col-md-1">&nbsp;</th>
+                        <th class="col-md-2">&nbsp;</th>
+                    </tr>
+                </thead>
+                <tbody>
+                    @foreach ($usersList as $luser)
```
Comment (nguyenthanhtung88) [inline resources/views/users/list.blade.php]: Why you use variable name `$luser`? It doesn't make sense, make your variable name easy to realize your logic.

File: resources/views/users/list.blade.php [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+                                {{ $luser->name }}
+                            </td>
+                            <td>
+                                {{ $luser->email }}
+                            </td>
+                            <td>
+                                &nbsp;
+                            </td>
+                            <td>
+                                @if (@in_array($luser->id,$follows))
```
Comment (nguyenthanhtung88) [inline resources/views/users/list.blade.php]: 1. lack space after `,`
2. Do you need use `@` for php function in_array
=>My suggestion: `@if (in_array($luser->id, $follows))`

**Comment by nguyenthanhtung88** [comment]
OK

**Pull request merged**